#!/usr/bin/env python2
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "asr_ism_visualizations"


gen = ParameterGenerator()

gen.add("arrowLifeTime", double_t, 0, "Time Markers are shown (0=permanent).", 0, 0, 120)
gen.add("pointLifeTime", double_t, 0, "Time Markers are shown (0=permanent).", 0, 0, 120)
gen.add("baseFrame", str_t, 0, "Base Frame of scene for poseprediction visualization.", "/map")
gen.add("referenceRadius", double_t, 0, "Width of lines for poseprediction visualization.", 0.025, 0, 0.02)
gen.add("pointRadius", double_t, 0, "Lenght of coodinateaxis for poseprediction visualization.", 0.02, 0, 0.2)
gen.add("arrowScaleX", double_t, 0, "Maximum distance [m] in overlay for poseprediction visualization.", 0.006, 0, 0.1)
gen.add("arrowScaleY", double_t, 0, "Maximum angle [DEG] in overlay for poseprediction visualization.", 0.006, 0, 0.1)
gen.add("arrowScaleZ", double_t, 0, "Maximum angle [DEG] in overlay for poseprediction visualization.", 0.006, 0, 0.1)
gen.add("colorStepSize", double_t, 0, "Maximum angle [DEG] in overlay for poseprediction visualization.", 30, 0, 360)


exit(gen.generate(PACKAGE, "asr_ism_visualizations", "rp_pose_prediction_visualizer"))
